Sources/Plasma/NucleusLib/pnUUID/pnUUID_Unix.cpp

```diff
@@ -50,41 +50,28 @@ static_assert(sizeof(plUUID) == sizeof(uuid_t), "plUUID and uuid_t types differ
 
 struct plUUIDHelper
 {
-    typedef struct _GUID {
-      uint32_t Data1;
-      uint16_t Data2;
-      uint16_t Data3;
-      uint8_t  Data4[8];
-    } GUID;
-    
-    static inline void CopyToPlasma( plUUID * dst, const uuid_t & src )
+    static inline void CopyToPlasma(plUUID* dst, const uuid_t& src)
     {
-        hsAssert( sizeof(uuid_t)==sizeof(dst->fData), "sizeof(uuid_t)!=sizeof(plUUID)" );
-        memcpy( (void*)dst->fData, (const void *)src, sizeof( plUUID ) );
-        
-        /*
-         GUIDs are always internally encoded in little endian,
-         while the uuid_t structure is big endian.
-         */
-        GUID* guid = reinterpret_cast<GUID*>(dst->fData);
-        guid->Data1 = hsSwapEndian32(guid->Data1);
-        guid->Data2 = hsSwapEndian16(guid->Data2);
-        guid->Data3 = hsSwapEndian16(guid->Data3);
+        memcpy(dst->fData, src, sizeof(plUUID));
+
+        // plUUIDs need to always be encoded in little endian, but the uuid_t struct is big endian
+        // This isn't guaranteed to be 32-bit-aligned, so casting to uint16_t/uint32_t pointers is UB
+        std::swap(dst->fData[0], dst->fData[3]);
+        std::swap(dst->fData[1], dst->fData[2]);
+        std::swap(dst->fData[4], dst->fData[5]);
+        std::swap(dst->fData[6], dst->fData[7]);
```

Comment:
While we're here, I wonder if we should add an unit test for uuids?

Reply:
Added some pnUUID unit tests, including one that specifically checks the endianness and passes both before and after this change